addons/sale_stock/models/sale_order.py

```diff
@@ -425,9 +410,10 @@ def create(self, vals_list):
     def write(self, values):
         lines = self.env['sale.order.line']
         if 'product_uom_qty' in values:
-            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
-            lines = self.filtered(
-                lambda r: r.state == 'sale' and not r.is_expense and float_compare(r.product_uom_qty, values['product_uom_qty'], precision_digits=precision) == -1)
+            for line in self:
+                if line.state == 'sale' and not line.is_expense:
+                    lines |= line
+
```

Comment:
It could be written in one line I think:
```suggestion
            lines |= self.filtered(lambda line: line.state == 'sale' and not line.is_expense)
```
But as you add a loop, maybe it worths it to include the `previous_product_uom_qty` in it ?
```python3
        if 'product_uom_qty' in values:
            previous_product_uom_qty = {}
            lines = self.env['sale.order.line']
            for line in self:
                if line.state == 'sale' and not line.is_expense:
                    lines |= line
                    previous_product_uom_qty[line.id] = line.product_uom_qty
```